pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java

```diff
@@ -43,21 +49,41 @@ public class DateTimeGranularitySpec {
    */
   public DateTimeGranularitySpec(String granularity) {
     Preconditions.checkArgument(StringUtils.isNotEmpty(granularity), "Must provide granularity");
-    String[] granularityTokens = StringUtil.split(granularity, SEPARATOR, 2);
-    Preconditions.checkArgument(granularityTokens.length >= NUM_TOKENS,
-        "Invalid granularity: %s, must be of format 'size:timeUnit", granularity);
+
+    int sizePosition;
+    int timeUnitPosition;
+    String[] granularityTokens;
+
+    if (Character.isDigit(granularity.charAt(0))) {
+      // Colon format
+      granularityTokens = StringUtil.split(granularity, COLON_SEPARATOR, 2);
+      Preconditions.checkArgument(granularityTokens.length >= NUM_TOKENS,
+              "Invalid granularity: %s, must be of format 'size:timeUnit", granularity);
```

Comment:
We can merge both the `Preconditions.checkArgument` behind a boolean flag and throw an error something like:
```suggestion
      Preconditions.checkArgument(granularityTokens.length >= NUM_TOKENS,
              "Invalid granularity: %s, must be of format 'size:timeUnit or timeUnit|size", granularity);
```